components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/DefaultPedigree.java

```diff
@@ -108,6 +110,47 @@ public static boolean isSupportedPedigreeFormat(JSONObject data)
         return extractedObjects;
     }
 
+    @Override
+    public void addLink(String patientId)
+    {
+        int maxExistingPedigreeID = 0;
+
+        // go through all members to find first unused ID, and to check if the patient
+        // is already in the pedigree
+        JSONArray members = (JSONArray) this.data.opt(PEDIGREE_JSON_MEMBERS_KEY);
+        for (Object nodeObj : members) {
+            JSONObject node = (JSONObject) nodeObj;
+
+            maxExistingPedigreeID = Math.max(maxExistingPedigreeID, node.optInt(PATIENT_NODEID_JSON_KEY, 0) + 1);
```

Comment:
Why the `+1`? If it's supposed to be the new id for the next patient, then the name is wrong, and if it's supposed to be the existing id, then incrementing here is wrong.